### PR TITLE
Update release-overview.md

### PR DESCRIPTION
**Pull request by amyblais**
Clarify release processes for:
1. Process for merging the ``master`` branch into the ``cloud`` branch
2. Process for cutting the release branch for Self-Managed releases based off of Cloud releases

### CODE REVIEW

File: operations/research-and-development/product/release-process/release-overview.md [hunk 1]
```diff
@@ -82,15 +82,24 @@ Release dates are currently communicated in the following ways. Further iteratio
 
 Release Manager is able to look at the version we have deployed to Mattermost Cloud. Also, when we are hooked up to split.io, there will be a dashboard where you can see active and historic feature flags. Any feature that has its flag removed would be included on any Self-Managed releases past the point after that removal was merged. More details on feature flags: https://developers.mattermost.com/contribute/server/feature-flags/.
 
+## Cloud Release Branch Processes
+
+**Process for merging the ``master`` branch into the ``cloud`` branch**
```
Comment (amyblais) [inline operations/research-and-development/product/release-process/release-overview.md:87]: @ethervoid My goal is to document this process for visibility for devs who may not know how it works yet. Let me know if any steps or details should be added below.

File: operations/research-and-development/product/release-process/release-overview.md [hunk 2]
```diff
@@ -82,15 +82,24 @@ Release dates are currently communicated in the following ways. Further iteratio
...
+ - The Cloud team opens PRs in webapp, server, redux, and enterprise branches to merge the ``master`` branch into the ``cloud`` branch ([example PR](https://github.com/mattermost/mattermost-server/pull/16838)).
+ - Devs should be aware of the dates when the ``master`` branch is merged into the ``cloud`` branch in order to be mindful of avoiding having incomplete features in a Cloud release and to be mindful of bug fixes that we may want to include in a release.
+
+**Process for cutting the release branch for Self-Managed releases based off of Cloud releases**
```
Comment (amyblais) [inline operations/research-and-development/product/release-process/release-overview.md:91]: @metanerd My goal is to document this process for visibility for devs who may not know how it works yet. Let me know if any steps or details should be added below.

Comment (metanerd) [reply operations/research-and-development/product/release-process/release-overview.md:91]: Looks good to me, thank you Amy!

File: operations/research-and-development/product/release-process/release-overview.md [hunk 3]
```diff
@@ -82,15 +82,24 @@ Release dates are currently communicated in the following ways. Further iteratio
 
 Release Manager is able to look at the version we have deployed to Mattermost Cloud. Also, when we are hooked up to split.io, there will be a dashboard where you can see active and historic feature flags. Any feature that has its flag removed would be included on any Self-Managed releases past the point after that removal was merged. More details on feature flags: https://developers.mattermost.com/contribute/server/feature-flags/.
 
+## Cloud Release Branch Processes
+
+**Process for merging the ``master`` branch into the ``cloud`` branch**
+ - The Cloud team opens PRs in webapp, server, redux, and enterprise branches to merge the ``master`` branch into the ``cloud`` branch ([example PR](https://github.com/mattermost/mattermost-server/pull/16838)).
```
Comment (ethervoid) [inline operations/research-and-development/product/release-process/release-overview.md]:
```suggestion
 - The Cloud team opens PRs in webapp, server, api-reference, redux, and enterprise branches to merge the ``master`` branch into the ``cloud`` branch ([example PR](https://github.com/mattermost/mattermost-server/pull/16838)).
```

**Review by ethervoid — DISMISSED** [review]
Despite that minor change all good! thank you!

File: operations/research-and-development/product/release-process/release-overview.md (hunk 3 above)
Comment (metanerd) [reply operations/research-and-development/product/release-process/release-overview.md]: Mmctl should also be in here!

Comment (ethervoid) [reply operations/research-and-development/product/release-process/release-overview.md]: @metanerd we're not doing cloud versioning for the mmctl project, should we? Given that we're using master always...

**Review by jespino — DISMISSED** [review]

File: operations/research-and-development/product/release-process/release-overview.md [hunk 4]
```diff
@@ -82,15 +82,24 @@ Release dates are currently communicated in the following ways. Further iteratio
...
+ - Devs should be aware of the dates when the ``master`` branch is merged into the ``cloud`` branch in order to be mindful of avoiding having incomplete features in a Cloud release and to be mindful of bug fixes that we may want to include in a release.
+
+**Process for cutting the release branch for Self-Managed releases based off of Cloud releases**
+ - Instead of using the tick-tock branching process, the Self-Managed releases are now cut based off of the Mattermost Cloud release tags (e.g Self-Managed v5.32 release was based off of ``cloud-2021-01-26`` tag).
+
 ## Adding Milestones on PRs and Jira Tickets
 
 Releases are now focused on "shipping features and improvements when they're ready for Mattermost Cloud, and then they'll get to a Self-Managed release once they've been available on Mattermost Cloud for 2+ weeks".
 
  - A new Cloud branch (based off of master) is used, and any regression bug fixes for the next Mattermost Cloud release will be cherry-picked there.
```
Comment (justinegeffen) [inline operations/research-and-development/product/release-process/release-overview.md]:
```suggestion
 - A new Cloud branch (based off of ``master``) is used, and any regression bug fixes for the next Cloud release will be cherry-picked there.
```

File: operations/research-and-development/product/release-process/release-overview.md [hunk 5]
```diff
@@ -82,15 +82,24 @@ Release dates are currently communicated in the following ways. Further iteratio
 
 Release Manager is able to look at the version we have deployed to Mattermost Cloud. Also, when we are hooked up to split.io, there will be a dashboard where you can see active and historic feature flags. Any feature that has its flag removed would be included on any Self-Managed releases past the point after that removal was merged. More details on feature flags: https://developers.mattermost.com/contribute/server/feature-flags/.
 
+## Cloud Release Branch Processes
+
+**Process for merging the ``master`` branch into the ``cloud`` branch**
+ - The Cloud team opens PRs in webapp, server, redux, api-reference, mmctl, and enterprise branches to merge the ``master`` branch into the ``cloud`` branch ([example PR](https://github.com/mattermost/mattermost-server/pull/16838)).
```
Comment (justinegeffen) [inline operations/research-and-development/product/release-process/release-overview.md]:
```suggestion
 - The Cloud team opens PRs in webapp, server, redux, api-reference, mmctl, and Enterprise branches to merge the ``master`` branch into the ``cloud`` branch ([example PR](https://github.com/mattermost/mattermost-server/pull/16838)).
```

File: operations/research-and-development/product/release-process/release-overview.md [hunk 6]
```diff
@@ -82,15 +82,24 @@ Release dates are currently communicated in the following ways. Further iteratio
...
  - This applies to webapp/server/Redux/Enterprise repos.
  - A fix version such as “Cloud (November 24)” is added in Jira to track regression bug fixes for Mattermost Cloud releases.
- - The Self-Managed releases will be based on Mattermost Cloud releases. This is decided by the Release Manager/Cloud team based on the timeline between Mattermost Cloud releases and the release day of Self-Managed releases.
- - Cloud PRs currently being merged into master don’t need a Self-Managed release milestone added to them. Release Manager will track merged PRs and help ensure that correct milestones are added as needed. Any concerns can be brought up for discussion on a case-by-case basis. There may be updates to this process in the future.
+ - The Self-Managed releases are based off of Mattermost Cloud releases.
+ - Submitted feature or bug fix PRs don’t need a release milestone added to them unless the PR is scheduled for a specific Mattermost Cloud or Self-Managed release. Release Manager will track merged PRs and help ensure that correct milestones are added as needed. Any concerns can be brought up for discussion on a case-by-case basis.
```
Comment (justinegeffen) [inline operations/research-and-development/product/release-process/release-overview.md]:
```suggestion
 - Submitted feature or bug fix PRs don’t need a release milestone added to them unless the PR is scheduled for a specific Mattermost Cloud or Self-Managed release. The Release Manager will track merged PRs and help ensure that correct milestones are added as needed. Any concerns can be brought up for discussion on a case-by-case basis.
```

**Review by justinegeffen — DISMISSED** [review]
Some minor comments but otherwise LGTM! :)

File: operations/research-and-development/product/release-process/release-overview.md [hunk 7]
```diff
@@ -82,15 +82,24 @@ Release dates are currently communicated in the following ways. Further iteratio
...
 
+## Cloud Release Branch Processes
+
+**Process for merging the ``master`` branch into the ``cloud`` branch**
+ - The Cloud team opens PRs in webapp, server, redux, api-reference, mmctl, and Enterprise branches to merge the ``master`` branch into the ``cloud`` branch ([example PR](https://github.com/mattermost/mattermost-server/pull/16838)).
+ - Devs should be aware of the dates when the ``master`` branch is merged into the ``cloud`` branch in order to be mindful of avoiding having incomplete features in a Cloud release and to be mindful of bug fixes that we may want to include in a release.
+
+**Process for cutting the release branch for Self-Managed releases based off of Cloud releases**
+ - Instead of using the tick-tock branching process, the Self-Managed releases are now cut based off of the Mattermost Cloud release tags (e.g Self-Managed v5.32 release was based off of ``cloud-2021-01-26`` tag).
```
Comment (amyblais) [inline operations/research-and-development/product/release-process/release-overview.md]: @jespino Did these details help understand the cloud release process a bit better? Do you think any other details could be added for clarification? We can also discuss in a DM if you'd like.

File: operations/research-and-development/product/release-process/release-overview.md (hunk 3 above)
Comment (amyblais) [reply operations/research-and-development/product/release-process/release-overview.md]: I've removed mmctl for now, but let me know if it should be re-added.

File: operations/research-and-development/product/release-process/release-overview.md [hunk 8]
```diff
@@ -82,15 +82,24 @@ Release dates are currently communicated in the following ways. Further iteratio
 
 Release Manager is able to look at the version we have deployed to Mattermost Cloud. Also, when we are hooked up to split.io, there will be a dashboard where you can see active and historic feature flags. Any feature that has its flag removed would be included on any Self-Managed releases past the point after that removal was merged. More details on feature flags: https://developers.mattermost.com/contribute/server/feature-flags/.
 
+## Cloud Release Branch Processes
+
+**Process for merging the ``master`` branch into the ``cloud`` branch**
+ - The Cloud team opens PRs in webapp, server, redux, api-reference, and Enterprise branches to merge the ``master`` branch into the ``cloud`` branch ([example PR](https://github.com/mattermost/mattermost-server/pull/16838)).
```
Comment (cwarnermm) [inline operations/research-and-development/product/release-process/release-overview.md:88]: Should "Enterprise" include a capital here? or should it be "enterprise"?

File: operations/research-and-development/product/release-process/release-overview.md [hunk 9]
```diff
@@ -82,15 +82,24 @@ Release dates are currently communicated in the following ways. Further iteratio
...
+ - Devs should be aware of the dates when the ``master`` branch is merged into the ``cloud`` branch in order to be mindful of avoiding having incomplete features in a Cloud release and to be mindful of bug fixes that we may want to include in a release.
```
Comment (cwarnermm) [inline operations/research-and-development/product/release-process/release-overview.md]:
```suggestion
 - Devs should be aware of the dates when a ``master`` branch is merged into a ``cloud`` branch in order to be mindful of avoiding having incomplete features in a Cloud release, and to include bug fixes that we may want to include in a release.
```

**Review by cwarnermm — COMMENTED** [review]
Minor feedback provided inline.

File: operations/research-and-development/product/release-process/release-overview.md (hunk 8 above)
Comment (amyblais) [reply operations/research-and-development/product/release-process/release-overview.md:88]: @cwarnermm I capitalized it based on Justine's suggestion https://github.com/mattermost/mattermost-handbook/pull/629#discussion_r582202940 - I'm 0/5.

**Review by cwarnermm — APPROVED** [review]
Reads well!

**Pull request merged**